lib/tasks/oauth_access_records.rake

```diff
@@ -14,7 +14,7 @@ namespace :oauth_access_records do
 
       until ids.empty?
         ids = klass
-          .where("expires_in is not null AND DATE_ADD(created_at, INTERVAL expires_in second) < ?", Time.zone.now)
+          .where("expires_in is not null AND DATE_ADD(created_at, INTERVAL expires_in second) <= ?", Time.zone.now)
```

Comment:
Minor: It might be nice to extract this query into a scope on a module mixed into both the relevant classes to make it more intention-revealing.

Reply:
> I can make this change with the scope and a mixin but because those models are owned by Doorkeeper, I think it would mean reopening `Doorkeeper::AccessGrant` and `Doorkeeper::AccessToken`. I was avoiding doing that, but maybe there's nothing wrong with it?
> 
> We do do it already for `Doorkeeper::Application` (https://github.com/alphagov/signon/blob/main/app/models/doorkeeper/application.rb)

Yes, I don't think there's a big danger in doing it as long as the new methods/scopes have distinct enough names, but I don't feel super strongly about it - I don't mind if you don't think it's worth the effort/risk.

Reply:
Stupid question: where should that mixin's file go?

Reply:
> Stupid question: where should that mixin's file go?

Given we don't have any already, I'd be inclined to just put it in `app/models` for now.

If you're not keen on that idea, you could consider making it an `ActiveSupport::Concern` and putting it in `app/models/concerns`...?

Reply:
On the off chance that it ever matters to anybody, I'll summarise the conclusion to this here. After a few attempts to mix-in the suggested mixin and only managing to raise uninitialised constant NameErrors, compounded with the confusion (and probably discomfort) of straddling/intruding on the Doorkeeper gem's own namespace, I wasn't sure how to resolve the errors and decided that it wasn't worth _that_ much effort anyway :smile: